File: day2/javascript/d02b.js
const fs = require('fs')
const readline = require('readline')

const fileLines = []

const fileInterface = readline.createInterface({
    input: fs.createReadStream('../../inputs/d02.txt')
})

fileInterface.on('line', line => {
    const [command, value] = line.split(" ")
    fileLines.push({ command, value: +value })
})

fileInterface.on('close', () => {
    const state = {
        horizontal: 0,
        depth: 0,
        aim: 0
    }

    fileLines.forEach(({ command, value }) => {
        switch (command) {
            case 'forward':
                state.horizontal += value
                state.depth += state.aim ? value * state.aim : 0
                break
            case 'up':
                state.aim -= value
                break
            case 'down':
                state.aim += value
        }
    })

    console.log(state, state.horizontal * state.depth)
})

File: day2/python/d02b.py
with open('./inputs/d02.txt', 'r') as file:
    data = [line.split(" ") for line in file.readlines()]


depth = 0
horizontal = 0
aim = 0

for command in data:
    if command[0] == 'forward':
        horizontal += int(command[1])
        depth += int(command[1]) * aim if aim > 0 else 0
    elif command[0] == 'up':
        aim -= int(command[1])
    elif command[0] == 'down':
        aim += int(command[1])

print(depth, horizontal, aim, depth * horizontal)
